contrib/registry/eureka/client.go

```diff
@@ -0,0 +1,356 @@
+	statusDown          = "DOWN"
+	statusOutOfServeice = "OUT_OF_SERVICE"
+	heartbeatRetry      = 3
+	maxIdleConns        = 100
+	heartbeatTime       = 10
+	httpTimeout         = 3
+)
+
+type Endpoint struct {
+	InstanceID     string
+	IP             string
+	AppID          string
+	Port           int
+	SecurePort     int
+	HomePageURL    string
+	StatusPageURL  string
+	HealthCheckURL string
+	MetaData       map[string]string
+}
+
+// Response for /eureka/apps
+type ApplicationsRootResponse struct {
+	ApplicationsResponse `json:"applications"`
+}
+
+type ApplicationsResponse struct {
+	Version      string        `json:"versions__delta"`
+	AppsHashcode string        `json:"apps__hashcode"`
+	Applications []Application `json:"application"`
+}
+
+type Application struct {
+	Name     string     `json:"name"`
+	Instance []Instance `json:"instance"`
+}
+
+type RequestInstance struct {
+	Instance Instance `json:"instance"`
+}
+
+type Instance struct {
+	InstanceID     string            `json:"instanceId"`
+	HostName       string            `json:"hostName"`
+	Port           Port              `json:"port"`
+	App            string            `json:"app"`
+	IPAddr         string            `json:"ipAddr"`
+	VipAddress     string            `json:"vipAddress"`
+	Status         string            `json:"status"`
+	SecurePort     Port              `json:"securePort"`
+	HomePageURL    string            `json:"homePageUrl"`
+	StatusPageURL  string            `json:"statusPageUrl"`
+	HealthCheckURL string            `json:"healthCheckUrl"`
+	DataCenterInfo DataCenterInfo    `json:"dataCenterInfo"`
+	Metadata       map[string]string `json:"metadata"`
+}
+
+type Port struct {
+	Port    int    `json:"$"`
+	Enabled string `json:"@enabled"`
+}
+
+type DataCenterInfo struct {
+	Name  string `json:"name"`
+	Class string `json:"@class"`
+}
+
+var _ APIInterface = new(Client)
+
+type APIInterface interface {
+	Register(ctx context.Context, ep Endpoint) error
+	Deregister(ctx context.Context, appID, instanceID string) error
+	Heartbeat(ep Endpoint)
+	FetchApps(ctx context.Context) []Application
+	FetchAllUpInstances(ctx context.Context) []Instance
+	FetchAppInstances(ctx context.Context, appID string) (m Application, err error)
+	FetchAppUpInstances(ctx context.Context, appID string) []Instance
+	FetchAppInstance(ctx context.Context, appID string, instanceID string) (m Instance, err error)
+	FetchInstance(ctx context.Context, instanceID string) (m Instance, err error)
+	Out(ctx context.Context, appID, instanceID string) error
+	Down(ctx context.Context, appID, instanceID string) error
+}
+
+type ClientOption func(e *Client)
+
+func WithMaxRetry(maxRetry int) ClientOption {
+	return func(e *Client) { e.maxRetry = maxRetry }
+}
+
+func WithHeartbeatInterval(interval string) ClientOption {
+	return func(e *Client) {
+		if duration := e.toDuration(interval); duration > 0 {
+			e.heartbeatInterval = duration
+		}
+	}
+}
+
+func WithCtx(ctx context.Context) ClientOption {
```

Comment:
WithContext()

Reply:
This package has the same nam with WithContext.
Rename WithClientContext.